src/RetroMole.Core/Interfaces/Texture.cs

```diff
@@ -7,21 +7,52 @@ public class Texture
 {
     public int Width;
     public int Height;
-    public byte[] Data;
     public IntPtr ID;
-    private Texture(string FilePath)
+
+    //------------------------------------------------------------------------------------------------------------------------
+    private Rgba32[] _Pixels;
+    public Rgba32 this[int x, int y]
+    {
+        get => _Pixels[x + y * Width];
+        set {
+            _Pixels[x + y * Width] = value;
+            OnChanged?.Invoke(new OnTextureChangedEventArgs(this));
+        }
+    }
+    public Rgba32[] Pixels
     {
-        Image<Bgra32> img = Image.Load<Bgra32>(FilePath);
-        this.Width = img.Width;
-        this.Height = img.Height;
+        get => _Pixels;
+        set {
+            _Pixels = _Pixels
+                .Select((p, i) => value[i])
+                .ToArray();
+            OnChanged?.Invoke(new OnTextureChangedEventArgs(this));
+        }
+    }
 
-        this.Data = new byte[this.Width * this.Height * 4];
-        img.CopyPixelDataTo(this.Data);
+    //----------------------------------------------------------------------------------------------------------------------
+    public event Action<OnTextureChangedEventArgs> OnChanged;
```

Comment:
Give this a default listener that calls `RetroMole.Core.GLOBALS.CurrentController.UpdateTexture(ID, Pixels)`